### PR TITLE
Add HTTP Client to opampextension

### PR DESCRIPTION
**Pull request by jaronoff97**
**Description:** <Describe what has changed.>

**Link to tracking Issue:** <Issue number if applicable>

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

**Review by evan-bradley — COMMENTED** [review]
Overall looks good so far to me.

**Comment by jaronoff97** [comment]
thanks! gonna add some tests and a changelog.

File: extension/opampextension/config_test.go [hunk 1]
```diff
@@ -41,20 +41,141 @@ func TestUnmarshalConfig(t *testing.T) {
...
+		wantErr assert.ErrorAssertionFunc
+	}{
+		{
+			name: "WS must have endpoint",
+			fields: fields{
+				Server: &OpAMPServer{
+					WS: &commonFields{},
+				},
+			},
+			wantErr: func(t assert.TestingT, err error, i ...any) bool {
```
Comment (evan-bradley) [inline extension/opampextension/config_test.go:82]:
```suggestion
			wantErr: func(t assert.TestingT, err error, _ ...any) bool {
```
I don't think these are used, right?

File: extension/opampextension/config_test.go [hunk 2]
```diff
@@ -41,20 +41,141 @@ func TestUnmarshalConfig(t *testing.T) {
...
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			cfg := &Config{
+				Server:       tt.fields.Server,
+				InstanceUID:  tt.fields.InstanceUID,
+				Capabilities: tt.fields.Capabilities,
+			}
+			tt.wantErr(t, cfg.Validate(), "Validate()")
```
Comment (evan-bradley) [inline extension/opampextension/config_test.go:178]:
```suggestion
			tt.wantErr(t, cfg.Validate())
```
Will this be used anywhere? Right now I don't think any of the functions make use of it.

**Review by evan-bradley — APPROVED** [review]

File: extension/opampextension/config_test.go (hunk 2 above)
Comment (jaronoff97) [reply extension/opampextension/config_test.go:178]: boilerplate from IDE 🙃

File: extension/opampextension/config_test.go (hunk 1 above)
Comment (jaronoff97) [reply extension/opampextension/config_test.go:82]: boilerplate from IDE 🙃

**Pull request merged**